mobilitypath_publisher/include/mobilitypath_publisher/mobilitypath_publisher.hpp

```diff
@@ -0,0 +1,162 @@
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <boost/shared_ptr.hpp>
+#include <lanelet2_extension/projection/local_frame_projector.h>
+
+#include <std_msgs/msg/string.hpp>
+#include <carma_planning_msgs/msg/trajectory_plan.hpp>
+#include <carma_v2x_msgs/msg/mobility_path.hpp>
+#include <carma_v2x_msgs/msg/bsm.hpp>
+
+#include <carma_ros2_utils/carma_lifecycle_node.hpp>
+#include "mobilitypath_publisher/mobilitypath_publisher_config.hpp"
+
+namespace mobilitypath_publisher
+{
+
+  /**
+   * \class MobilityPathPublication
+   * \brief The class responsible for publishing MobilityPath messages based on the latest
+   * trajectory plan 
+   * 
+   */
+  class MobilityPathPublication : public carma_ros2_utils::CarmaLifecycleNode
+  {
+
+  private:
+    // Subscribers
+    carma_ros2_utils::SubPtr<carma_planning_msgs::msg::TrajectoryPlan> traj_sub_;
+    carma_ros2_utils::SubPtr<carma_v2x_msgs::msg::BSM> bsm_sub_;
+    carma_ros2_utils::SubPtr<std_msgs::msg::String> georeference_sub_;
+
+    // Publishers
+    carma_ros2_utils::PubPtr<carma_v2x_msgs::msg::MobilityPath> path_pub_;
+
+    // Timer for publishing MobilityPath message
+    rclcpp::TimerBase::SharedPtr path_pub_timer_;
+
+    // Node configuration
+    Config config_;
+
+    // The most recently published trajectory plan
+    carma_planning_msgs::msg::TrajectoryPlan latest_trajectory_;
+
+    // The MobilityPath message generated from the most recently published trajectory plan
+    carma_v2x_msgs::msg::MobilityPath latest_mobility_path_;
+
+    // The BSMCoreData from the most recently published BSM message 
+    carma_v2x_msgs::msg::BSMCoreData bsm_core_;
+
+    // Map projection string, which defines the lat/lon -> map conversion
+    std::shared_ptr<lanelet::projection::LocalFrameProjector> map_projector_;
+
+    // Recipient's static ID (Empty string indicates a broadcast message)
+    std::string recipient_id = "";
+
+    // Sender's dynamic ID which is its BSM id in hex string
+    std::string sender_bsm_id = "FFFF";
+
+    /**
+    * \brief Spin callback, which will be called frequently based on the configured spin rate
+    */
+    bool spin_callback();
+
+    /**
+     * \brief Callback for the trajectory plan subscriber, which stores the latest trajectory plan locally
+     * and stores its corresponding MobilityPath locally
+     * \param msg Latest trajectory plan message
+     */
+    void trajectory_cb(const carma_planning_msgs::msg::TrajectoryPlan::UniquePtr msg);
+
+    /**
+     * \brief Callback for the BSM subscriber, which stores the BSM's BSMCoreData locally
+     * \param msg Latest BSM message
+     */
+    void bsm_cb(const carma_v2x_msgs::msg::BSM::UniquePtr msg);
+
+    /**
+     * \brief Generates a MobilityHeader to be used for a published MobilityPath message
+     * \param time Time in milliseconds
+     * \return A MobilityHeader 
+     */
+    carma_v2x_msgs::msg::MobilityHeader composeMobilityHeader(uint64_t time);
```

Comment:
Lets make all the functions have the same snake_case name format to keep the it consistent within the package.